### PR TITLE
STACKDEV-64 : SkyWalking Production Deployment on Azure Cloud

### PR DESCRIPTION
**Pull request by Keshav8s**
This pull request is to merge changes from the `skywalking-azure` branch in [Keshav8s forked repo](https://github.com/Keshav8s/skywalking) into the newly created branch `skywalking-azure` in the main [cloudscaleinc repo](https://github.com/cloudscaleinc/skywalking) repository.

### CODE REVIEW

File: .github/workflows/deploy-azure.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          ${{ runner.os }}-buildx-
+
+    - name: Deploy to Azure VM
+      env:
+        SSH_PRIVATE_KEY_BASE64: ${{ secrets.SSH_PRIVATE_KEY_BASE64 }}
+        VM_IP: ${{ secrets.VM_IP }}
+        VM_USER: ${{ secrets.VM_USER }}
+      run: |
+        mkdir -p ~/.ssh
+        echo "$SSH_PRIVATE_KEY_BASE64" | base64 --decode > ~/.ssh/id_ed25519
```
Comment (ritesh-cloudscaleinc) [inline .github/workflows/deploy-azure.yml]: why do we need key at every execution ? Also can this be securely fetched from env variables ?

File: .github/workflows/publish-docker-e2e-service.yaml [hunk 2]
```diff
@@ -1,36 +1,19 @@
-# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (ritesh-cloudscaleinc) [inline .github/workflows/publish-docker-e2e-service.yaml]: we should not remove any licensed or reference to open source

File: .github/workflows/publish-docker-e2e-service.yaml [hunk 3]
```diff
@@ -1,36 +1,19 @@
...
-      - master
-    paths:
-      - 'test/e2e-v2/java-test-service/**'
-      - 'test/Makefile'
+      - skywalking-azure
 
 env:
   SKIP_TEST: true
-  HUB: ghcr.io/apache/skywalking
+  HUB: ghcr.io/keshav8s/skywalking
```
Comment (ritesh-cloudscaleinc) [inline .github/workflows/publish-docker-e2e-service.yaml]: this needs update- use cloudscaleinc/skywalking

File: .github/workflows/publish-docker-e2e-service.yaml [hunk 4]
```diff
@@ -1,36 +1,19 @@
...
 
 jobs:
   build:
-    if: github.repository == 'apache/skywalking'
+    if: github.repository == 'Keshav8s/skywalking'
```
Comment (ritesh-cloudscaleinc) [inline .github/workflows/publish-docker-e2e-service.yaml]: this need update see my previous comment

File: .gitmodules [hunk 5]
```diff
@@ -9,4 +9,4 @@
 	url = https://github.com/apache/skywalking-data-collect-protocol.git
 [submodule "skywalking-ui"]
 	path = skywalking-ui
-	url = https://github.com/apache/skywalking-booster-ui.git
+	url = git@github.com:Keshav8s/skywalking-booster-ui.git
```
Comment (ritesh-cloudscaleinc) [inline .gitmodules]: +1 need update

File: README.md [hunk 6]
```diff
@@ -1,69 +1,88 @@
-Apache SkyWalking
```
Comment (ritesh-cloudscaleinc) [inline README.md]: don;t remove any reference to original README including any other references. Create another README name it was README-CloudScale and add reference to main file

**Review by ritesh-cloudscaleinc — CHANGES_REQUESTED** [review]
need update refer to reveiw comments

File: .github/workflows/deploy-azure.yml (hunk 1 above)
Comment (Keshav8s) [reply .github/workflows/deploy-azure.yml]: Without keys i didn't found any other ways to connect with the vm in order to make any changes we need authentication. Yes we can use for env variables as well, under secrets and variables.

File: .github/workflows/publish-docker-e2e-service.yaml (hunk 2 above)
Comment (Keshav8s) [reply .github/workflows/publish-docker-e2e-service.yaml]: okay !

File: README.md (hunk 6 above)
Comment (Keshav8s) [reply README.md]: okay !

File: .github/workflows/publish-docker-e2e-service.yaml (hunk 3 above)
Comment (Keshav8s) [reply .github/workflows/publish-docker-e2e-service.yaml]: yes, i have updated the yaml file

File: .github/workflows/deploy-azure.yml (hunk 1 above)
Comment (ritesh-cloudscaleinc) [inline .github/workflows/deploy-azure.yml]: this needs update ~/.ssh/id_ed25519

File: .github/workflows/deploy-azure.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        chmod 600 ~/.ssh/id_ed25519
+        eval $(ssh-agent -s)
+        ssh-add ~/.ssh/id_ed25519
```
Comment (ritesh-cloudscaleinc) [inline .github/workflows/deploy-azure.yml]: +1

File: docker/docker-compose.yml [hunk 8]
```diff
@@ -67,7 +67,7 @@ services:
     networks:
       - demo
     healthcheck:
-      test: [ "CMD-SHELL", "curl http://localhost:12800/internal/l7check" ]
+      test: ["CMD-SHELL", "curl --silent --fail http://localhost:12800/internal/l7check || exit 1"]
```
Comment (ritesh-cloudscaleinc) [inline docker/docker-compose.yml]: why this check example - silent fail ?

**Review by ritesh-cloudscaleinc — COMMENTED** [review]
few minor updates reqd

File: .github/workflows/deploy-azure.yml (hunk 1 above)
Comment (Keshav8s) [reply .github/workflows/deploy-azure.yml]: updated this line

File: docker/docker-compose.yml (hunk 8 above)
Comment (Keshav8s) [reply docker/docker-compose.yml]: This was made when ui was not working, will edit this again !

**Review by ritesh-cloudscaleinc — APPROVED** [review]
PR is not in sync

**Review by ritesh-cloudscaleinc — APPROVED** [review]
LGTM

**Pull request closed without merging**